### PR TITLE
Seal key

### PR DESCRIPTION
**Pull request by msenmurugan**

### CODE REVIEW

File: pkg/apps/blackduck/latest/creater.go [hunk 1]
```diff
@@ -153,13 +91,31 @@ func (hc *Creater) Ensure(blackduck *v1.Blackduck) error {
...
 	errors := commonConfig.CRUDComponents()
 	if len(errors) > 0 {
-		return fmt.Errorf("unable to update components due to %+v", errors)
+		return fmt.Errorf("unable to update non postgres and uploadcache components due to %+v", errors)
 	}
 
+	// log.Debugf("created/updated non postgres and upload cache component for %s", blackduck.Spec.Namespace)
+
+	// deploy upload cache component
+	commonConfig = crdupdater.NewCRUDComponents(hc.KubeConfig, hc.KubeClient, hc.Config.DryRun, blackduck.Spec.Namespace,
```
Comment (jdartigalongue) [inline pkg/apps/blackduck/latest/creater.go]: Why aren't we deploying uploadcache and everything else (except postgres) at the same time?

Comment (msenmurugan) [reply pkg/apps/blackduck/latest/creater.go]: this is bcoz, if there is any change in seal key, it need to restart only the upload cache container

File: pkg/apps/blackduck/latest/deployer.go [hunk 2]
```diff
@@ -25,55 +25,77 @@ import (
...
 	"github.com/blackducksoftware/horizon/pkg/components"
 	horizon "github.com/blackducksoftware/horizon/pkg/deployer"
-
 	"github.com/blackducksoftware/synopsys-operator/pkg/api"
 	"github.com/blackducksoftware/synopsys-operator/pkg/api/blackduck/v1"
-	bdutil "github.com/blackducksoftware/synopsys-operator/pkg/blackduck/util"
-	"github.com/blackducksoftware/synopsys-operator/pkg/util"
-
 	containers "github.com/blackducksoftware/synopsys-operator/pkg/apps/blackduck/latest/containers"
+	hubutils "github.com/blackducksoftware/synopsys-operator/pkg/blackduck/util"
```
[the remark this answers is not in the document]
Comment (msenmurugan) [reply pkg/apps/blackduck/latest/deployer.go]: renamed to bdutils

File: pkg/apps/blackduck/latest/deployer.go [hunk 3]
```diff
@@ -25,55 +25,77 @@ import (
...
+
+	// Get the flavor
+	flavor, err := hc.getContainersFlavor(blackduck)
+	if err != nil {
+		return nil, err
+	}
+
 	containerCreater := containers.NewCreater(hc.Config, &blackduck.Spec, flavor)
 
+	// Persistent Volume Claim
```
[the remark this answers is not in the document]
Comment (msenmurugan) [reply pkg/apps/blackduck/latest/deployer.go]: commented the patch and remove logic for pvc

File: pkg/apps/blackduck/latest/containers/postgres.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	} else {
+		hubDbConfig.AddData(map[string]string{"HUB_POSTGRES_ENABLE_SSL": "false"})
+	}
+	hubDbConfig.AddLabels(c.GetVersionLabel("postgres"))
+
+	return hubDbConfig
+}
+
+// GetPostgresSecret will return the postgres secret
+func (c *Creater) GetPostgresSecret(adminPassword string, userPassword string) *components.Secret {
```
[the remark this answers is not in the document]
Comment (msenmurugan) [reply pkg/apps/blackduck/latest/containers/postgres.go]: fixed

File: pkg/blackduck/ctl_blackduck.go [hunk 5]
```diff
@@ -303,3 +305,17 @@ func (ctl *Ctl) SetFlag(f *pflag.Flag) {
...
+
+// SpecIsValid verifies the spec has necessary fields to deploy
+func (ctl *Ctl) SpecIsValid() (bool, error) {
+	return true, nil
+}
+
+// CanUpdate checks if a user has permission to modify based on the spec
+func (ctl *Ctl) CanUpdate() (bool, error) {
+	version := util.GetHubVersion(ctl.Spec.Environs)
+	if version == "" || version == "2019.2.2" {
```
[the remark this answers is not in the document]
Comment (msenmurugan) [reply pkg/blackduck/ctl_blackduck.go]: @mphammer will fix this issue in another PR

**Pull request merged**